elixir-europe-training/Training-Metrics-Database
# Feat/add user accessible data import

This PR tries to implement feature #36

## To test
- `docker compose up`
- go to: http://localhost:8000
- sign in as any node
- go to the tab [Upload data](http://localhost:8000/upload-data)
- upload your data one file at a time
- go to the tab [Edit events](http://localhost:8000/event/list)
- verify that you can edit your events
- verify that the event stats properly reflects what you uploaded

## Test dataset
A test data set `test-data.zip` which contains three example files. One for each type. In order to make the test data set work in your environment you must first upload the events and then map the rows of each metric entry to the event codes returned by the event upload.

[test-data.zip](https://github.com/user-attachments/files/15874052/test-data.zip)


## Note
There are some issues still remaining with this implementation when it comes to importing data entries from the csv files.

- How should `code` be assigned to imported events?
- How should `code` of imported events be communicated to the user?
- What is the `status` of an imported event?
- What is the `duration` of an imported event?
- In order for the system to work properly I assume that the field `code` needs a uniqueness requirement?
